simpleclient_hotspot/src/main/java/io/prometheus/client/hotspot/StandardExports.java

```diff
@@ -101,6 +99,72 @@ public StandardExports() {
     return mfs;
   }
 
+  static Long callLongGetter(String getterName, Object obj) throws NoSuchMethodException {
+    Method method = obj.getClass().getMethod(getterName);
+    return (Long) callMethod(method, obj).getValue();
+  }
+
+  /**
+   * Attempts to call a method either directly or via one of the implemented interfaces.
+   */
+  static ReturnValue callMethod(Method method, Object obj) {
+    try {
+      method.setAccessible(true);
```

Comment:
The call itself should work just fine. All the Java 9 [AccessibleObject.setAccessible()](http://download.java.net/java/jdk9/docs/api/java/lang/reflect/AccessibleObject.html#setAccessible-boolean-) documentation says in addition to the previous versions is that there is some logic preventing this working across module boundaries unless there is a direct module dependency and the method is public (essentially). Seeing how the two UnixOperatingSystemMXBeans are interfaces, the methods are definitely public. So in the worst case one would need to define a module dependency to make this work. Which, if necessary at all, would need to be defined for any kind of implementation to work.

To summarize, if any code will work in Java 9 then this should work too.